calculator.rb

```diff
@@ -0,0 +1,158 @@
+#Calculator Program: Week1 day4
+
+#METHODS
+def add(a, b)
+  return "#{a} + #{b} = #{a + b}"
+end
+
+
+def subtract(a, b)
+  return "#{a} - #{b} = #{a - b}"
+end
+
+
+def multiply(a, b)
+  return "#{a} * #{b} = #{a * b}"
+end
+
+
+def divide(a, b)
+  a = a.to_f
+  b = b.to_f
+  if b == 0
+    puts "Undefined."
+  else
+    return "#{a} / #{b} = #{a / b}"
+  end
+end
+
+def mod(a, b)
+  return "#{a} % #{b} = #{a % b}"
+end
+
+def exp(a, b)
+  return "#{a} ** #{b} = #{a ** b}"
+end
+
+
+def f_or_i(number)
+  if number.include?(".")
+    return number.to_f
+  else
+    return number.to_i
+  end
+end
+
+def splitting(input)
+  return input.split(" ")
+end
+
+def compare(newness, original)
+  if newness.to_s != original
+    puts "EEP! I'm going to *gracefully* convert your string to a 0 and multiply that."
+  end
+end
+#if I can get the the parenthesis contained in one place
+
+
+#USER INTERFACE
+puts "Enter your equation. You must put spaces between numbers and operations."
+operation = gets.chomp
+
+until operation.include?("add") || operation.include?("+") || operation.include?("sub") ||
+  operation.include?("-") || operation.include?("mul") || operation.include?("*") ||
+  operation.include?("divide") || operation.include?("/") || operation.include?("%") ||
+  operation.include?("mod") || operation.include?("**") || operation.include?("exp") || operation.include?("power")
+  puts "Sorry, invalid operation. Please input your equation again."
+  operation = gets.chomp
+end
+
+if operation.include?(" ")
+  operation_modified = splitting(operation)
```

Comment:
the names `operation` and `operation_modified` is a little confusing to me since they represent the whole input expression so I didn't understand what it meant until I read through